File: Practice/AllDomains/CoreCS/DataStructures/Arrays/LeftRotation.cs
/*	author:@shivkrthakur  */
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution {
    static void Main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        int [] nd = Console.ReadLine().Split().Select(Int32.Parse).ToArray();
        int n = nd[0], d = nd[1];
        int [] input = Console.ReadLine().Split().Select(Int32.Parse).ToArray();

        int [] output = new int[n];
        for(int i = 0; i < n; i++)
        {
            int t = (i + d) % n;
            output[i] = input[t];
            //Console.WriteLine(t);
        }
        Console.WriteLine(string.Join(" ",output));
    }
}

File: Practice/AllDomains/CoreCS/DataStructures/Arrays/LeftRotation.js
/*	author:@shivkrthakur  */
var index = 0;

function readLine(input){
    return input[index++];
}

function processData(input) {
    //Enter your code here
    var ns = readLine(input).split(' ').map(Number);
    var arr = readLine(input).split(' ').map(Number);
    var n = ns[0], s = ns[1];

    var opArr = [];
    for(var i = 0; i < n; i++){
        opArr[i] = arr[(i + s) % n];
    }
    console.log(opArr.join(' '));
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input.split('\n'));
});
